runtime/include/QuantumDevice.hpp

```diff
@@ -35,16 +35,18 @@
 namespace Catalyst::Runtime {
 
 /**
- * @brief struct API for backend quantum devices.
+ * @brief Interface class for Catalyst Runtime device backends.
  *
- * This device API contains,
- * - a set of methods to manage qubit allocations and deallocations, device shot
- *   noise, and quantum tape recording as well as reference values for the result
- *   data-type; these are used to implement Quantum Runtime (QR) instructions.
- *
- * - a set of methods for quantum operations, observables, measurements, and gradient
- *   of the device; these are used to implement Quantum Instruction Set (QIS) instructions.
+ * All Catalyst device plugins must implement this class and distribute it as a shared library.
+ * See https://docs.pennylane.ai/projects/catalyst/en/stable/dev/custom_devices.html for details.
  *
+ * The QuantumDevice interface methods can broadly be categorized into device management functions
```

Comment:
PL & Lightning use backtick for the name of classes and functions in docstrings. What's the style here?

For example,
```suggestion
 * The `QuantumDevice` interface methods can broadly be categorized into device management functions
```

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1680/commits/e7c92a7ead42d8ac4d000addedfa0cfa5c69405f